ext/mbstring/mbstring.c

```diff
@@ -2626,12 +2633,13 @@ static zend_string* mb_trim_string(zend_string *input, zend_string *marker, cons
 		if (out_len <= from) {
 			from -= out_len;
 		} else {
-			for (unsigned int i = from; i < out_len; i++) {
-				width -= character_width(wchar_buf[i]);
-				if (width < 0) {
+			for (size_t i = from; i < out_len; i++) {
+				size_t current_wchar_char_width = character_width(wchar_buf[i]);
+				if (width < current_wchar_char_width) {
 					enc->from_wchar(wchar_buf + from, i - from, &buf, true);
 					goto append_trim_marker;
 				}
+				width -= character_width(wchar_buf[i]);
```

Comment:
I think this should be `width -= current_wchar_char_width;`, not so?

Reply:
Yes indeed, I will fix this and rebase the fix into the commit